ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,377 @@
+    <dataset id>.<annex uuid of the remote>.log showing 'who' ran into this issue with what dataset. Since this logging
+    can potentially leak personal data (like local file paths for example) it can be disabled from the client side via
+    `annex.ria-remote.<RIASIBLING>.ignore-remote-config`.
+
+    Todo
+    ----
+    Where to put the description of a RIA store (see below)?
+
+    The targeted layout of such a store is a tree of datasets, starting at the configured base path. First level of
+    subdirectories are named for the first three characters of the datasets' id, second level is the remainder of those
+    ids. The thereby created dataset directories contain a bare git repository.
+    Those bare repositories are slightly different from plain git-annex bare repositories in that they use the standard
+    dirhashmixed layout beneath annex/objects as opposed to dirhashlower, which is git-annex's default for bare
+    repositories. Furthermore, there is an additional directory 'archives' within the dataset directories, which may or
+    may not contain archives with annexed content.
+    Note, that this helps to reduce the number of inodes consumed (no checkout + potential archive) as well as it allows
+    to resolve dependencies (that is (sub)datasets) merely by their id.
+    Finally, there is a file `ria-layout-version` put beneath the store's base path, determining the version of the
+    dataset tree layout and a file of the same name per each dataset directory determining object tree layout version
+    (we already switch from dirhashlower to dirhashmixed for example) and an additional directory `error_logs` at the
+    toplevel.
+    """
+
+    # TODO: option to skip existing remotes in case of recursive?
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to process.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        url=Parameter(
+            args=("url",),
+            metavar="URL",
+            doc="""""",
+            constraints=EnsureStr() | EnsureNone()),
+        name=Parameter(
+            args=('-s', '--name',),
+            metavar='NAME',
+            doc="""sibling name to create for this publication target.
+                If `recursive` is set, the same name will be used to label all
+                the subdatasets' siblings. When creating a target dataset fails,
+                no sibling is added""",
+            constraints=EnsureStr() | EnsureNone()),
+        ria_sibling=Parameter(
+            args=("--ria-sibling",),
+            metavar="RIASIBLING",
+            doc="""name of the RIA storage sibling (git-annex special remote). Must not be identical to NAME. 
+            By default NAME is appended with '-ria'""",
```

Comment:
What about calling it `--ria-remote-name`? Rational: the dataset sibling is actually identified with `--name`, so being close to that makes sense. OTOH the RIA remote is a key store, and less of an actual "sibling", because it doesn't deal with a repo clone at all (unlike the Git remote). Having it be called `--ria-sibling` sounds like there would be two siblings, but there is only one.

Reply:
True, but I guess we don't have a convention yet on how to talk about special remotes in datalad. What term do I use within docstrings then? "remote"? That seems equally confusing.

Reply:
I think they are "git annex special remotes", but we don't need to invent a grant scheme IMHO.